content/articles/building-a-lyrics-search-app-with-vanilla-js/index.md

```diff
@@ -0,0 +1,472 @@
+
+Since the data has been mapped, we access the song title using `song.title` and song artist using `song.artist.name`.
+
+The following code helps us listen to click event inside the output innerHTML:
+
+```javascript
+//event listener in get lyrics button
+output.addEventListener('click', e=>{
+    const clickedElement = e.target;
+
+    //checking clicked element is button or not
+    if (clickedElement.tagName === 'SPAN'){
+        const artist = clickedElement.getAttribute('data-artist');
+        const songTitle = clickedElement.getAttribute('data-songtitle');
+        
+        getLyrics(artist, songTitle)
+    }
+})
+
+```
+
+Above is the code that helps to know if the clicked element is the span tag. If it is, (i.e the span tag is === variable clicked button), the statement in the if block is executed because the span tag bears the tagname `SPAN`. We store this attribute in variable `songTiltle` and also the attribute for the song artist in variable `artist` which the `getLyrics()` function.
+
+### GetLyrics() async function
+This is the most important function because this is what makes us get the lyrics themselves displayed. The function takes in two arguments artist and songTitle respectively, recall that you can search for lyrics through inputting song artist or song title. This is the reason why the getLyrics function is taking those two arguments.
+
+Below is the code for getLyrics async function:
+
+```javascript
+async function getLyrics(artist, songTitle) {
+    const response = await fetch(`${api}/v1/${artist}/${songTitle}`);
+    const data = await response.json();
+  
+    const lyrics = data.lyrics;
+    if (lyrics === undefined){
+        alert('lyrics doesnt exist in this api');
+        console.log('lyrics doesnt exist in this api');
+    }
+  
+    output.innerHTML = `<h2><strong>${artist}</strong> - ${songTitle}</h2>
+    <p id="lyrics-display">${lyrics}</p>`;
+  
+  }
+```
+Note, the api doesn't have all the lyrics of songs it suggests, it's still in progress, if you wish to have an app that can return so many lyrics, i will suggest you get a paid lyrics api. The lyrics that are not avialable, returns undefined in the div to display lyrics, it also alerts you that the lyrics is not available on the api. Also, when you input a song title to search from, you have to exercise little patience for the data to load because it is a bit slow.
+
+Next, you can decide to implement the regular expression syntax in the variable lyrics using the `replace` method. For well-aligned lyrics, the carriage returns, and line breaks are removed and replaced with a <br>. The regular expressions help to look through for any carriage return alone or carriage return with a new line and replace it with the <br>. 
+
+The `/g` is a regular expression flag, it means globally, the entire document to be searched through(all matches).
```

Comment:
you do not have the regular expressions code in the article so it might be a little confusing when you talk about it without a frame of reference

Reply:
@shegz101 
have you addressed this?